### PR TITLE
Fixes #2853

### LINKED ISSUE #2853: Vector Setting Error
The following input file raises the error: `Cannot set element 12 since there are only 12 elements.` Somewhere, Psi is trying to set a non-existing vector element. I'm very curious why tests didn't catch this.

```
memory 2 GB

molecule {
0 1
C           1.741722       -1.200108       -0.733703
C           1.723351       -1.235440        0.673033
C           1.741208       -0.034341        1.406196
C           1.720430        1.201594        0.733472
C           1.738484        1.236058       -0.673301
C           1.720301        0.035400       -1.407211
H           1.811924       -2.125930       -1.299462
H           1.810840       -2.187236        1.191603
H           1.810814       -0.061176        2.490913
H           1.805598        2.126733        1.298598
H           1.806257        2.188975       -1.192470
H           1.805027        0.062415       -2.491005
--
2 1
Ba         -1.306490       -0.000406        0.000195
}
basis {
assign  jun-cc-pVDZ
assign Ba GENECP

[GENECP]
cartesian
****
Ba     0 
S   3   1.00
      0.8699000             -2.2549747        
      0.6676000              2.5145786        
      0.1982000              0.5775184        
S   4   1.00
      0.8699000              2.0391383        
      0.6676000             -2.3717712        
      0.1982000             -1.2758006        
      0.0823000              1.1703346        
S   1   1.00
      0.0231000              1.0000000        
P   3   1.00
      1.6050000             -0.1626403        
      0.4790000              0.6971289        
      0.1818000              0.4505107        
P   2   1.00
      0.1804000             -0.2642537        
      0.0476000              1.1347212        
P   1   1.00
      0.0192000              1.0000000 
****
}

energy('sapt0')
```

### PR DESCRIPTION
**Pull request by JonathonMisiewicz**
## User API & Changelog headlines

- [x] Provided a more informative error message when SAD crashes due to an atom having more core electrons than basis functions.

## Dev notes & details

- [x] Fixed typos in comments
- [x] Fixed buggy print information

## Questions
- [ ] Is it worth testing for the error message explicitly?

## Checklist
- [x] Confirmed I get the expected error message upon running the example from #2853.

## Status
- [x] Ready for review
- [x] Ready for merge upon @susilehtola approval

### CODE REVIEW

**Review by zachglick — APPROVED** [review]

File: psi4/src/psi4/libscf_solver/sad.cc [hunk 1]
```diff
@@ -288,6 +288,9 @@ void SADGuess::run_atomic_calculations(SharedMatrix& DAO, SharedMatrix& HuckelC,
             // Sanity check: can't have more active orbitals than basis functions
             if (nact > nbf - nfzc) {
                 nact = nbf - nfzc;
+                if (nact < 0) {
+                    throw PSIEXCEPTION("SAD: Atom " + molecule_->symbol(index) + " has more core electrons than basis functions.");
+                }
```
Comment (susilehtola) [inline psi4/src/psi4/libscf_solver/sad.cc:293]: This is the wrong place
```suggestion
```

File: psi4/src/psi4/libscf_solver/sad.cc [hunk 2]
```diff
@@ -254,7 +254,7 @@ void SADGuess::run_atomic_calculations(SharedMatrix& DAO, SharedMatrix& HuckelC,
```
Comment (susilehtola) [inline psi4/src/psi4/libscf_solver/sad.cc:254]: This is where it should be 
```suggestion
if(nelec[index]>2*nbf) {
throw PSIEXCEPTION("SAD: Atom " + molecule_->symbol(index) + " has more core electrons than basis functions.");
}
```

**Comment by JonathonMisiewicz** [comment]
I modified the error message from being about "core electrons" to just "electrons." Aside from that, @susilehtola's comments addressed.

**Review by maxscheurer — APPROVED** [review]

**Comment by susilehtola** [comment]
> I modified the error message from being about "core electrons" to just "electrons." 

Ah, copy-paste error. You didn't accept the suggestions tho

**Comment by JonathonMisiewicz** [comment]
I wrote and pushed my own commit implementing them (fixing the copy-paste error). See the latest commit. I don't see an issue here.

**Comment by JonathonMisiewicz** [comment]
> I wrote and pushed my own commit implementing them (fixing the copy-paste error). See the latest commit. I don't see an issue here.

@susilehtola

**Comment by susilehtola** [comment]
> > I wrote and pushed my own commit implementing them (fixing the copy-paste error). See the latest commit. I don't see an issue here.

In the future, please use "accept suggestion".

**Comment by JonathonMisiewicz** [comment]
> > > I wrote and pushed my own commit implementing them (fixing the copy-paste error). See the latest commit. I don't see an issue here.
> 
> In the future, please use "accept suggestion".

Not when I need to suggest a change to the suggestion.

**Pull request merged**